### PR TITLE
Django 1.9 boilerplate

### PR DESCRIPTION
**Pull request by m-kalyna**
w/ Gunicorn role

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Byte-compiled / optimized / DLL files
+# Byte-1ompiled / optimized / DLL files
```
Comment (darthwade) [inline .gitignore]: what?

File: .gitignore [hunk 2]
```diff
@@ -87,3 +88,6 @@ ENV/
 
 # Rope project settings
 .ropeproject
+
+#IDE
+.idea
```
Comment (darthwade) [inline .gitignore]: Please clear .gitignore

File: ansible/files/local_settings.py.j2 [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+        'HOST': 'localhost',
+        'PORT': '5432',
+    }
+}
+
+DATABASES['default']['NAME'] = '{{ environment_variables.DATABASE_NAME }}'
+DATABASES['default']['USER'] = '{{ environment_variables.DATABASE_USER }}'
+DATABASES['default']['PASSWORD'] = '{{ environment_variables.DATABASE_PASSWORD }}'
+DATABASES['default']['HOST'] = '{{ environment_variables.DATABASE_HOST }}'
+DATABASES['default']['PORT'] = {{ environment_variables.DATABASE_PORT }}
```
Comment (darthwade) [inline ansible/files/local_settings.py.j2]: not good, override it above

File: ansible/files/local_settings.py.j2 [hunk 4]
```diff
@@ -0,0 +1,51 @@
+DEBUG = {{ environment_variables.DEBUG }}
+
+# Database
+DATABASES = {
+    'default': {
+        'ENGINE': 'django.contrib.gis.db.backends.postgis',
```
Comment (darthwade) [inline ansible/files/local_settings.py.j2]: postgis?

File: ansible/group_vars/all.yml [hunk 5]
```diff
@@ -0,0 +1,403 @@
+---
+### Vault ###
+vaulted_database_name: 'boilerplate'
+vaulted_database_user: 'boilerplate'
+vaulted_database_password: 'password'
```
Comment (darthwade) [inline ansible/group_vars/all.yml]: use vault

File: ansible/group_vars/all.yml [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+postgresql_users:
+  - name: '{{ vaulted_database_user }}'
+    pass: '{{ vaulted_database_password }}'
+    encrypted: no # denotes if the password is already encrypted.
+
+postgresql_user_privileges:
+  - name: '{{ vaulted_database_user }}'
+    db: '{{ vaulted_database_name }}'
+    priv: 'ALL'
+    role_attr_flags: 'SUPERUSER'
```
Comment (darthwade) [inline ansible/group_vars/all.yml]: why superuser?

File: ansible/group_vars/all.yml [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+celery_run:
+  - action: worker
+    loglevel: info
+
+  - action: beat
+    loglevel: info
+
+### Gunicorn ###
+# Installation
+gunicorn_version: " b"
```
Comment (darthwade) [inline ansible/group_vars/all.yml]: b ?

File: ansible/group_vars/all.yml [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+gunicorn_app_name: "{{ deploy_app_name }}"
+gunicorn_config: "{{ deploy_etc_dir }}/{{ gunicorn_app_name }}_gunicorn.py"
+gunicorn_virtualenv: "{{ virtualenv_dir }}"
+# Running
+gunicorn_bind: "unix:{{ deploy_run_dir }}/boilerplate.sock"
+gunicorn_workers: 8
```
Comment (darthwade) [inline ansible/group_vars/all.yml]: 8 ? 
you have 8 cores? :)

File: ansible/roles/docs/templates/docs.conf.j2 [hunk 9]
```diff
@@ -0,0 +1,23 @@
+#!upstart
```
Comment (darthwade) [inline ansible/roles/docs/templates/docs.conf.j2]: upstart? 👎

File: ansible/roles/ssh/defaults/main.yml [hunk 10]
```diff
@@ -0,0 +1,7 @@
+---
+ssh_user: '{{ deploy_user }}'
+ssh_group: '{{ deploy_group }}'
+ssh_dir: '/home/{{ ssh_user }}/.ssh'
+ssh_known_hosts: []
+ssh_private_key: "{{ lookup('template', playbook_dir + '/files/ssh/key') }}"
+ssh_public_key: "{{ lookup('template', playbook_dir + '/files/ssh/key.pub') }}"
```
Comment (darthwade) [inline ansible/roles/ssh/defaults/main.yml]: I think we need remove this role and operate keys manually

File: ansible/roles/docs/templates/docs.conf.j2 (hunk 9 above)
Comment (m-kalyna) [inline ansible/roles/docs/templates/docs.conf.j2]: @darthwade waitin' for circusCI role :) Or better use supervisord?

File: ansible/group_vars/all.yml (hunk 6 above)
Comment (m-kalyna) [inline ansible/group_vars/all.yml]: @darthwade i'm dunno. Does it matter?

File: ansible/group_vars/all.yml (hunk 8 above)
Comment (Wildream) [inline ansible/group_vars/all.yml]: 8 workers is a good count for production, I guess, but default value should be ~2 .

File: ansible/roles/docs/templates/docs.conf.j2 (hunk 9 above)
Comment (Wildream) [inline ansible/roles/docs/templates/docs.conf.j2]: We'll replace all upstart scripts with single circus config later.

File: ansible/group_vars/all.yml (hunk 8 above)
Comment (m-kalyna) [inline ansible/group_vars/all.yml]: Now it's 2

File: ansible/host_vars/production.yml [hunk 11]
```diff
@@ -1,38 +1,76 @@
...
-
-
-### Domains ###
-site_domain: ''
-api_domain: 'api.{{ site_domain }}'
-flower_domain: 'flower.{{ site_domain }}'
-tornado_domain: 'ws.{{ site_domain }}'
-docs_domain: 'docs.{{ site_domain }}'
\ No newline at end of file
+$ANSIBLE_VAULT;1.1;AES256
```
Comment (Wildream) [inline ansible/host_vars/production.yml]: It doesn't make sense to use ansible vault at all for this template project. All sensitive variables should be changed anyway when building a new project based on this project template.

File: ansible/roles/webtier/tasks/api.yml [hunk 12]
```diff
@@ -0,0 +1,84 @@
...
+#  tags: [provision, api]
+
+- name: Django | Collect static assets
+  django_manage:
+    command: collectstatic
+    settings: '{{ django_settings }}'
+    app_path: '{{ django_dir }}'
+    virtualenv: '{{ virtualenv_dir }}'
+  tags: [provision, api]
+  notify:
```
Comment (Wildream) [inline ansible/roles/webtier/tasks/api.yml]: This should be untied from static assets collection and performed as separate task. You can use `command: 'true'` in task config in order to stop Ansible from complaining about an empty task.

File: ansible/roles/gunicorn/handlers/main.yml [hunk 13]
```diff
@@ -0,0 +1,4 @@
+---
+- name: Restart gunicorn
+  become: true
+  service: name={{ gunicorn_app_name }}_gunicorn state=restarted
```
Comment (Wildream) [inline ansible/roles/gunicorn/handlers/main.yml]: please use dash (-) instead of underscore in a service name.

File: boilerplate/urls.py [hunk 14]
```diff
@@ -0,0 +1,13 @@
+from django.conf import settings
+from django.conf.urls import url
+from django.contrib import admin
+from django.conf.urls.static import static
+
+
+urlpatterns = [
+    url(r'^admin/', admin.site.urls),
```
Comment (Wildream) [inline boilerplate/urls.py]: This urls file misses some config for social auth app which we have installed by default. Please add it:
```
    url(r'^social/', include('social.apps.django_app.urls', namespace='social')),
    url(r'^auth/', include('rest_framework_social_oauth2.urls')),
```

File: boilerplate/settings.py [hunk 15]
```diff
@@ -0,0 +1,228 @@
+import os
+
+# Build paths inside the project like this: os.path.join(BASE_DIR, ...)
+from celery.schedules import crontab
```
Comment (Wildream) [inline boilerplate/settings.py]: Do we need this line here? This module is imported, but not used anywhere.

Comment (m-kalyna) [inline boilerplate/settings.py]: This is for configuring periodic celery task. Whatever, can remove it.

**Comment by Wildream** [comment]
@darthwade: I'd suggest to merge this PR now and fix the remaining issues later, since a lot of work depends on this initial PR, and considering how slow the reviewing process is now, we'll have to wait for eternity before we can start adding more features.

**Pull request merged**